dist/comment_style.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+
+import argparse, os, re, subprocess, sys
+
+def print_msg(file_name, line_num, line, multiline):
+    print("Illegal " + multiline + "comment in " + file_name + ":" + str(line_num - 1) + " " +
+        line.strip('\n'))
+
+def check_c_comments(file_name):
+    count = 0
+    with open(file_name) as f:
+        line_num = 0
+        for line in f:
+            stripped = line.strip()
+            if match(stripped):
+                print_msg(file_name, line_num, line, "")
+                count +=1
+            line_num += 1
+    return count
+
+def match(stripped_line):
+    # We don't worry about whitespace in matches as we strip the line of whitespace.
+    url = re.compile(r'https?:\/\/')
+
+    # // Style comments
+    single_line_comment = re.compile(
+        r'^(?:[^"/\\]|\"(?:[^\"\\]|\\.)*\"|/(?:[^/"\\]|\\.)|/\"(?:[^\"\\]|\\.)*\"|\\.)*//(.*)$')
+
+    return single_line_comment.match(stripped_line) and not url.search(stripped_line)
+
+def check_cpp_comments(file_name):
+    # Text before comments
+    text_check = re.compile(r'^[^\/\n]+')
+
+    with open(file_name) as f:
+        count = 0
+        line_num = 0
+        length = 0
+        for line in f:
+            stripped = line.strip()
+            if match(stripped):
+                # Detecting multi-line comments of // is not easy as technically they can occur
+                # on contiguous lines without being multi-line.
+                # E.g:
+                # int height; // the height of our object
+                # int width; // the width of our object
+                # So we can't just count the number of lines we need to check that there is no text
+                # preceding the comment in subsequent lines.
+                if length != 0 and not text_check.match(line.strip()):
+                    # If the comment is length 2 and we found another matching line then we have
+                    # found an illegal comment.
+                    if length == 2:
+                        print_msg(file_name, line_num, line, "multiline ")
+                        count += 1
+                    length += 1
+                    # Try and print only one error per comment, just keep incrementing the count and
+                    # the above if will only run once.
+                else:
+                    length += 1
+            else:
+                length = 0
+            line_num += 1
+        return count
+
+def file_is_cpp(name):
+    if re.search('(.cpp|.cxx|.hpp)$', name) is not None:
```

Comment:
We don't have any `.hpp` files so we can drop that extension

Reply:
I get it, but I'd prefer to leave it in. It's a reasonable choice for future code to add, and leaving it out will cause hard-to-spot issues. There's also no real cost to its presence here.